packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -292,7 +286,14 @@ class TransactionService {
   }
 
   startSpan(name, type, options) {
-    const tr = this.ensureCurrentTransaction()
+    const tr = this.ensureCurrentTransaction(
+      undefined,
+      TEMPORARY_TYPE,
+      this.createOptions({
```

Comment:
I would move these logic inside the function and get rid of passing around parameters here. WDYT? Would reduce some uncessary code for `startSpan` and addTask.

Reply:
`ensureCurrentTransaction` replaces `createTransaction` which we had before. I think this way it makes `ensureCurrentTransaction` more useful in different scenarios where we don't want to create options. see the tests for examples.

Reply:
I was also saying this because, it would reduce the code size. If we can come up with a way to make both work, it would be better.